main/templates/main/messages.html

```diff
@@ -13,13 +13,30 @@ <h5 class="d-inline">Messages</h5>
     <div class="card-body">
       {% now "e (O)" as server_time %}
       <span id="timezone" class="d-line text-body-tertiary">All timestamps are displayed in {{ server_time }}</span>
-      <input class="form-control mb-2"
-             type="search"
-             name="search"
-             placeholder="Search messages..."
-             hx-get="{% url 'main:messages' topic|default:"ALL" %}"
-             hx-trigger="input changed delay:500ms, every 1s"
-             hx-target="#message-list">
+      <!-- Filter Form -->
+      <form id="filter-form" method="get">
+        <!-- Search Input -->
+        <input class="form-control mb-2"
+               type="search"
+               name="search"
+               placeholder="Search messages..."
+               hx-get="{% url 'main:messages' topic|default:"ALL" %}"
+               hx-trigger="input changed delay:500ms, every 1s"
+               hx-target="#message-list"
+               hx-include="#filter-form">
+        <!-- Severity Filter Dropdown -->
+        <select name="severity"
```

Comment:
Far from being an expert in HTML, you should be able to select multiple options here by adding the `multiple` attribute. See https://www.w3schools.com/TAGS/att_select_multiple.asp Now idea what the choice look like in the backend, but you should be able to handle al the required logic there.

Reply:
Ok, it won't work exactly as a dropdown, but can be tweaked a bit. See https://stackoverflow.com/a/65969350/3778792

Reply:
If we go this route then [Shoelace dropdowns](https://shoelace.style/components/dropdown) support having checkboxes. We'd need to figure out how it interacts with HTMX though.